File: src/app/shared/constants/login-color-shape-set.ts
/**
 * Representa una combinacion de color y forma para login visual.
 */
export interface ColorShape {
  /** Codigo unico de la combinacion (ej: RED_CIRCLE) */
  code: string;
  /** Color en formato hexadecimal (ej: #F44336) */
  hexColor: string;
  /** Nombre de la forma en espanol (ej: Circulo) */
  shapeName: string;
  /** Tipo de forma para renderizado */
  shapeType: ShapeType;
  /** Nombre del color en espanol (ej: Rojo) */
  colorName: string;
  /** Nombre para mostrar al usuario (ej: Circulo Rojo) */
  displayName: string;
}

/** Tipos de formas disponibles */
export type ShapeType = 'circle' | 'square' | 'triangle' | 'star';

/** Colores disponibles */
export type ShapeColor = 'RED' | 'BLUE' | 'GREEN' | 'YELLOW' | 'ORANGE' | 'PURPLE';

/**
 * Set de colores y formas disponibles para login visual.
 * 24 combinaciones: 6 colores x 4 formas.
 */
export const LoginColorShapeSet = {
  items: [
    // Rojos
    { code: 'RED_CIRCLE', hexColor: '#F44336', shapeName: 'Circulo', shapeType: 'circle' as ShapeType, colorName: 'Rojo', displayName: 'Circulo Rojo' },
    { code: 'RED_SQUARE', hexColor: '#F44336', shapeName: 'Cuadrado', shapeType: 'square' as ShapeType, colorName: 'Rojo', displayName: 'Cuadrado Rojo' },
    { code: 'RED_TRIANGLE', hexColor: '#F44336', shapeName: 'Triangulo', shapeType: 'triangle' as ShapeType, colorName: 'Rojo', displayName: 'Triangulo Rojo' },
    { code: 'RED_STAR', hexColor: '#F44336', shapeName: 'Estrella', shapeType: 'star' as ShapeType, colorName: 'Rojo', displayName: 'Estrella Roja' },

    // Azules
    { code: 'BLUE_CIRCLE', hexColor: '#2196F3', shapeName: 'Circulo', shapeType: 'circle' as ShapeType, colorName: 'Azul', displayName: 'Circulo Azul' },
    { code: 'BLUE_SQUARE', hexColor: '#2196F3', shapeName: 'Cuadrado', shapeType: 'square' as ShapeType, colorName: 'Azul', displayName: 'Cuadrado Azul' },
    { code: 'BLUE_TRIANGLE', hexColor: '#2196F3', shapeName: 'Triangulo', shapeType: 'triangle' as ShapeType, colorName: 'Azul', displayName: 'Triangulo Azul' },
    { code: 'BLUE_STAR', hexColor: '#2196F3', shapeName: 'Estrella', shapeType: 'star' as ShapeType, colorName: 'Azul', displayName: 'Estrella Azul' },

    // Verdes
    { code: 'GREEN_CIRCLE', hexColor: '#4CAF50', shapeName: 'Circulo', shapeType: 'circle' as ShapeType, colorName: 'Verde', displayName: 'Circulo Verde' },
    { code: 'GREEN_SQUARE', hexColor: '#4CAF50', shapeName: 'Cuadrado', shapeType: 'square' as ShapeType, colorName: 'Verde', displayName: 'Cuadrado Verde' },
    { code: 'GREEN_TRIANGLE', hexColor: '#4CAF50', shapeName: 'Triangulo', shapeType: 'triangle' as ShapeType, colorName: 'Verde', displayName: 'Triangulo Verde' },
    { code: 'GREEN_STAR', hexColor: '#4CAF50', shapeName: 'Estrella', shapeType: 'star' as ShapeType, colorName: 'Verde', displayName: 'Estrella Verde' },

    // Amarillos
    { code: 'YELLOW_CIRCLE', hexColor: '#FFEB3B', shapeName: 'Circulo', shapeType: 'circle' as ShapeType, colorName: 'Amarillo', displayName: 'Circulo Amarillo' },
    { code: 'YELLOW_SQUARE', hexColor: '#FFEB3B', shapeName: 'Cuadrado', shapeType: 'square' as ShapeType, colorName: 'Amarillo', displayName: 'Cuadrado Amarillo' },
    { code: 'YELLOW_TRIANGLE', hexColor: '#FFEB3B', shapeName: 'Triangulo', shapeType: 'triangle' as ShapeType, colorName: 'Amarillo', displayName: 'Triangulo Amarillo' },
    { code: 'YELLOW_STAR', hexColor: '#FFEB3B', shapeName: 'Estrella', shapeType: 'star' as ShapeType, colorName: 'Amarillo', displayName: 'Estrella Amarilla' },

    // Naranjas
    { code: 'ORANGE_CIRCLE', hexColor: '#FF9800', shapeName: 'Circulo', shapeType: 'circle' as ShapeType, colorName: 'Naranja', displayName: 'Circulo Naranja' },
    { code: 'ORANGE_SQUARE', hexColor: '#FF9800', shapeName: 'Cuadrado', shapeType: 'square' as ShapeType, colorName: 'Naranja', displayName: 'Cuadrado Naranja' },
    { code: 'ORANGE_TRIANGLE', hexColor: '#FF9800', shapeName: 'Triangulo', shapeType: 'triangle' as ShapeType, colorName: 'Naranja', displayName: 'Triangulo Naranja' },
    { code: 'ORANGE_STAR', hexColor: '#FF9800', shapeName: 'Estrella', shapeType: 'star' as ShapeType, colorName: 'Naranja', displayName: 'Estrella Naranja' },

    // Violetas
    { code: 'PURPLE_CIRCLE', hexColor: '#9C27B0', shapeName: 'Circulo', shapeType: 'circle' as ShapeType, colorName: 'Violeta', displayName: 'Circulo Violeta' },
    { code: 'PURPLE_SQUARE', hexColor: '#9C27B0', shapeName: 'Cuadrado', shapeType: 'square' as ShapeType, colorName: 'Violeta', displayName: 'Cuadrado Violeta' },
    { code: 'PURPLE_TRIANGLE', hexColor: '#9C27B0', shapeName: 'Triangulo', shapeType: 'triangle' as ShapeType, colorName: 'Violeta', displayName: 'Triangulo Violeta' },
    { code: 'PURPLE_STAR', hexColor: '#9C27B0', shapeName: 'Estrella', shapeType: 'star' as ShapeType, colorName: 'Violeta', displayName: 'Estrella Violeta' },
  ] as ColorShape[],

  /** Longitud de la secuencia de login (4 elementos) */
  SEQUENCE_LENGTH: 4,

  /** Cantidad de elementos a mostrar en la grilla de seleccion */
  DISPLAY_COUNT: 9,

  /** Total de combinaciones disponibles */
  get totalCount(): number {
    return this.items.length;
  },
};

File: InclusiON.Shared/Constants/LoginColorShapeSet.cs
namespace InclusiON.Shared.Constants
{
    /// <summary>
    /// Set de colores y formas disponibles para login visual.
    /// 24 combinaciones: 6 colores x 4 formas.
    /// </summary>
    public static class LoginColorShapeSet
    {
        /// <summary>
        /// Todas las combinaciones de color y forma disponibles.
        /// </summary>
        public static readonly ColorShape[] Items = new[]
        {
            // Rojos
            new ColorShape("RED_CIRCLE", "#F44336", "Circulo", "circle", "Rojo"),
            new ColorShape("RED_SQUARE", "#F44336", "Cuadrado", "square", "Rojo"),
            new ColorShape("RED_TRIANGLE", "#F44336", "Triangulo", "triangle", "Rojo"),
            new ColorShape("RED_STAR", "#F44336", "Estrella", "star", "Rojo"),

            // Azules
            new ColorShape("BLUE_CIRCLE", "#2196F3", "Circulo", "circle", "Azul"),
            new ColorShape("BLUE_SQUARE", "#2196F3", "Cuadrado", "square", "Azul"),
            new ColorShape("BLUE_TRIANGLE", "#2196F3", "Triangulo", "triangle", "Azul"),
            new ColorShape("BLUE_STAR", "#2196F3", "Estrella", "star", "Azul"),

            // Verdes
            new ColorShape("GREEN_CIRCLE", "#4CAF50", "Circulo", "circle", "Verde"),
            new ColorShape("GREEN_SQUARE", "#4CAF50", "Cuadrado", "square", "Verde"),
            new ColorShape("GREEN_TRIANGLE", "#4CAF50", "Triangulo", "triangle", "Verde"),
            new ColorShape("GREEN_STAR", "#4CAF50", "Estrella", "star", "Verde"),

            // Amarillos
            new ColorShape("YELLOW_CIRCLE", "#FFEB3B", "Circulo", "circle", "Amarillo"),
            new ColorShape("YELLOW_SQUARE", "#FFEB3B", "Cuadrado", "square", "Amarillo"),
            new ColorShape("YELLOW_TRIANGLE", "#FFEB3B", "Triangulo", "triangle", "Amarillo"),
            new ColorShape("YELLOW_STAR", "#FFEB3B", "Estrella", "star", "Amarillo"),

            // Naranjas
            new ColorShape("ORANGE_CIRCLE", "#FF9800", "Circulo", "circle", "Naranja"),
            new ColorShape("ORANGE_SQUARE", "#FF9800", "Cuadrado", "square", "Naranja"),
            new ColorShape("ORANGE_TRIANGLE", "#FF9800", "Triangulo", "triangle", "Naranja"),
            new ColorShape("ORANGE_STAR", "#FF9800", "Estrella", "star", "Naranja"),

            // Violetas
            new ColorShape("PURPLE_CIRCLE", "#9C27B0", "Circulo", "circle", "Violeta"),
            new ColorShape("PURPLE_SQUARE", "#9C27B0", "Cuadrado", "square", "Violeta"),
            new ColorShape("PURPLE_TRIANGLE", "#9C27B0", "Triangulo", "triangle", "Violeta"),
            new ColorShape("PURPLE_STAR", "#9C27B0", "Estrella", "star", "Violeta"),
        };

        /// <summary>
        /// Longitud de la secuencia de login (4 elementos).
        /// </summary>
        public const int SequenceLength = 4;

        /// <summary>
        /// Cantidad de elementos a mostrar en la grilla de seleccion.
        /// </summary>
        public const int DisplayCount = 9;

        /// <summary>
        /// Total de combinaciones disponibles.
        /// </summary>
        public static int TotalCount => Items.Length;
    }

    /// <summary>
    /// Representa una combinacion de color y forma para login visual.
    /// </summary>
    public class ColorShape
    {
        /// <summary>
        /// Codigo unico de la combinacion (ej: RED_CIRCLE).
        /// </summary>
        public string Code { get; }

        /// <summary>
        /// Color en formato hexadecimal (ej: #F44336).
        /// </summary>
        public string HexColor { get; }

        /// <summary>
        /// Nombre de la forma en espanol (ej: Circulo).
        /// </summary>
        public string ShapeName { get; }

        /// <summary>
        /// Tipo de forma para renderizado (circle, square, triangle, star).
        /// </summary>
        public string ShapeType { get; }

        /// <summary>
        /// Nombre del color en espanol (ej: Rojo).
        /// </summary>
        public string ColorName { get; }

        /// <summary>
        /// Nombre para mostrar al usuario (ej: Circulo Rojo).
        /// </summary>
        public string DisplayName => $"{ShapeName} {ColorName}";

        public ColorShape(string code, string hexColor, string shapeName, string shapeType, string colorName)
        {
            Code = code;
            HexColor = hexColor;
            ShapeName = shapeName;
            ShapeType = shapeType;
            ColorName = colorName;
        }
    }
}
